lib/src/status.rs

```diff
@@ -305,19 +307,141 @@ pub(crate) async fn status(opts: super::cli::StatusOpts) -> Result<()> {
+                    .unwrap_or_else(|| "No timestamp present".to_owned());
+                let transport = &image.image.transport;
+                let digest = &image.image_digest;
+
+                writeln!(out, "    Image version: {version} ({timestamp})")?;
+                writeln!(out, "    Image transport: {transport}")?;
+                writeln!(out, "    Image digest: {digest}")?;
+            } else {
+                writeln!(out, "No {status_string} image present, checksum defined")?;
+            }
+        } else {
+            writeln!(out, "No {status_string} image present")?;
+        }
+    }
+    Ok(())
+}
+
+#[test]
+fn test_human_readable_base_spec() {
+    // Tests Staged and Booted, null Rollback
+    let spec_fixture: &str = include_str!("fixtures/spec-staged-booted.yaml");
+    let host: Host = serde_yaml::from_str(spec_fixture).unwrap();
+    let mut w = Vec::new();
+    human_readable_output(&mut w, &host).unwrap();
+    let w = String::from_utf8(w).unwrap();
+    dbg!(&w);
+    let expected = indoc::indoc! { r"
+    Current staged image: quay.io/example/someimage:latest
+        Image version: nightly (2023-10-14 19:22:15 UTC)
+        Image transport: registry
+        Image digest: sha256:16dc2b6256b4ff0d2ec18d2dbfb06d117904010c8cf9732cdb022818cf7a7566
+    Current booted image: quay.io/example/someimage:latest
+        Image version: nightly (2023-09-30 19:22:16 UTC)
+        Image transport: registry
+        Image digest: sha256:736b359467c9437c1ac915acaae952aad854e07eb4a16a94999a48af08c83c34
+    No rollback image present
+    "};
+    similar_asserts::assert_eq!(w, expected);
+}
+
+#[test]
+fn test_human_readable_rfe_spec() {
+    // Basic rhel for edge bootc install with nothing
+    let spec_fixture = include_str!("fixtures/spec-rfe-ostree-deployment.yaml");
+    let host = serde_yaml::from_str(spec_fixture).unwrap();
+    let mut w = Vec::new();
+    human_readable_output(&mut w, &host).unwrap();
+    let w = String::from_utf8(w).unwrap();
+    dbg!(&w);
+    let expected = indoc::indoc! { r"
+    No staged image present, checksum defined
+    No booted image present, checksum defined
+    No rollback image present
+    "};
+    similar_asserts::assert_eq!(w, expected);
+}
+
+#[test]
+fn test_human_readable_staged_spec() {
+    // staged image, no boot/rollback
+    let spec_fixture: &str = include_str!("fixtures/spec-ostree-to-bootc.yaml");
+    let host: Host = serde_yaml::from_str(spec_fixture).unwrap();
+    let mut w = Vec::new();
+    human_readable_output(&mut w, &host).unwrap();
+    let w = String::from_utf8(w).unwrap();
+    dbg!(&w);
+    let expected = indoc::indoc! { r"
+    Current staged image: quay.io/centos-bootc/centos-bootc:stream9
+        Image version: stream9.20240807.0 (No timestamp present)
+        Image transport: registry
+        Image digest: sha256:47e5ed613a970b6574bfa954ab25bb6e85656552899aa518b5961d9645102b38
+    No booted image present, checksum defined
```

Comment:
We could probably phrase this in a way more understandable to the user. One may very well feel that they are in a "booted image", just not a *container* image right?

Hmm...how about: "Booted state is native ostree" ?

We could also actually go a step farther and detect the situation of rpm-ostree generated content, though that'd be a good bit more work.

Reply:
Updated the output line to match what you suggested, if you'd like me to dig in further let me know